src/module-loader/dotnet/tests/MorganStanley.ComposeUI.ModuleLoader.Tests/AggregateModuleCatalogTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+using FluentAssertions;
+
+namespace MorganStanley.ComposeUI.ModuleLoader.Tests;
+
+public class AggregateModuleCatalogTests
+{
+    private readonly IModuleCatalog _moduleCatalog = new AggregateModuleCatalog(
+        new[]
+        {
+            new MockModuleCatalog(new[]
+            {
+                new MockModuleManifest("testModuleId", "testModuleName"),
+                new MockModuleManifest("testModuleId1", "testModuleName1"),
+                new MockModuleManifest("testModuleId2", "testModuleName2"),
+                new MockModuleManifest("testModuleId3", "testModuleName3"),
+                new MockModuleManifest("testModuleId4", "testModuleName4")
+            }),
+            new MockModuleCatalog(new[]
+            {
+                new MockModuleManifest("testModuleId5", "testModuleName5"),
+                new MockModuleManifest("testModuleId", "testModuleName8"),
+                new MockModuleManifest("testModuleId6", "testModuleName6"),
+            }),
+            new MockModuleCatalog(new[]
+            {
+                new MockModuleManifest("testModuleId", "testModuleName9"),
+            }),
+            new MockModuleCatalog(new[]
+            {
+                new MockModuleManifest("testModuleId7", "testModuleName7"),
+            }),
+            new MockModuleCatalog(Enumerable.Empty<MockModuleManifest>()),
+        });
+
+    [Theory]
+    [ClassData(typeof(AggregateModuleCatalogReturnManifestTheoryData))]
+    public async Task GetManifest_ReturnsModuleManifest_WhenModuleIdPassed(string moduleId, IModuleManifest expectedModuleManifest)
+    {
+        var result = await _moduleCatalog.GetManifest(moduleId);
+        result.Should().NotBeNull();
+        result.Should().BeEquivalentTo(expectedModuleManifest);
+    }
+
+    [Theory]
+    [InlineData("noModuleId")]
+    [InlineData("testModuleId8")]
+    public async Task GetManifest_ThrowsModuleNotFoundException_WhenNoModuleIdFound(string moduleId)
+    {
+        var action = () => _moduleCatalog.GetManifest(moduleId);
+        await action.Should().ThrowAsync<ModuleNotFoundException>();
+    }
+
+    [Theory]
+    [InlineData("testModuleId")]
```

Comment:
[ZKRobi](https://github.com/ZKRobi) [4 days ago](https://github.com/morganstanley/ComposeUI/pull/569#discussion_r1555999304)
The same Id exists with the same name multiple times. This doesn't verify the original source for the manifest as it can return any of the occurances and the assertion will stay true.

Reply:
Resolved as it was resolved on the original PR.